packages/button/index.js

```diff
@@ -0,0 +1,99 @@
+import { html, LitElement } from 'lit';
+import { classNames } from '@chbphone55/classnames';
+import { FabricElement } from '../utils';
+
+const variantClassMap = {
+  primary: 'button button--primary',
+  secondary: 'button',
+  negative: 'button button--destructive',
+  utility: 'button button--utility',
+  link: 'button button--link',
+  pill: 'button button--pill',
+};
+
+class FabricButton extends FabricElement {
+  static shadowRootOptions = { ...LitElement.shadowRootOptions, delegatesFocus: true };
+
+  static properties = {
+    type: { type: 'button' | 'submit' | 'reset', reflect: true },
+    autofocus: { type: Boolean, reflect: true },
+    variant: { type: String, reflect: true },
+    quiet: { type: Boolean, reflect: true },
+    small: { type: Boolean, reflect: true },
+    loading: { type: Boolean, reflect: true },
+    href: { type: String, reflect: true },
+    target: { type: String, reflect: true },
+    rel: { type: String, reflect: true },
+    className: { type: String, reflect: true },
+  };
+
+  constructor() {
+    super();
+    this.variant = 'secondary';
+  }
+
+  connectedCallback() {
+    super.connectedCallback();
+
+    const availableVariants = Object.keys(variantClassMap);
+    if (!availableVariants.includes(this.variant)) {
+      throw new Error(
+        `Invalid "variant" attribute. Set its value to one of the following:\n${availableVariants.join(
+          ', ',
+        )}.`,
+      );
+    }
+  }
+
+  firstUpdated() {
+    if (this.autofocus) {
+      setTimeout(() => this.focus(), 0);
+    }
+  }
+
+  get _classes() {
+    return classNames(
+      variantClassMap[this.variant],
+      {
+        // quiet
+        'button--flat': this.variant === 'secondary' && this.quiet,
+        'button--destructive-flat': this.variant === 'negative' && this.quiet,
+        'button--utility-flat': this.variant === 'utility' && this.quiet,
+        // others
+        'button--small': this.small,
+        'button--in-progress': this.loading,
+      },
+      this.className,
```

Comment:
I wonder, do we need this at all?
Since the component is an actual element, they can set class="" on the element and those styles will be applied to the element which should get the job done in many cases I would think. Are there concrete use cases you can think of where we would want the user to be able to set styles explicitly on the button element itself you think?

Reply:
You're right! It seems most style adjustments can be done at the custom element level with the use of `class` attribute. Removed `className` property in [16719cb](https://github.com/fabric-ds/elements/pull/95/commits/16719cbda9c6f8cd292a2a245dff1946c57020a7)